Remove attachment_filters guarded action

oliverguenther opened the pull request:
https://community.openproject.org/projects/openproject/work_packages/46797/activity

oliverguenther committed:
Remove attachment_filters guarded action

dombesz committed:
Remove the advanced filter ee feature reference from the docs

dombesz committed:
Remove full text search ee addon doc reference

dombesz commented on docs/user-guide/search/README.md:
```diff
@@ -38,6 +38,3 @@ The search bar of Openproject is configured to allow a full text search. This me
 The title and the description of a work package can also be found by [filtering by text](../work-packages/work-package-table-configuration/#filtering-by-text).
 Aside from full-text search, you can search work packages based on their attached file name or even within the content of an attachment.
-
-**Note** Full text search is an Enterprise add-on in OpenProject and can only be used with [Enterprise cloud](../../enterprise-guide/enterprise-cloud-guide/) or [Enterprise on-premises](../../enterprise-guide/enterprise-on-premises-guide/). An upgrade from the free Community edition is easily possible.
```
@oliverguenther I think this also needs to be removed as there are no other full text search filters that are EE filter.

dombesz committed:
Fix query-filters.component eslint error

dombesz reviewed (approved):
Apart from the failing test, the changes are LGTM.

dombesz committed:
Improve the Components::WorkPackage::Filter#expect_available_filter method to search for the option we are testing.

This way we can be sure that the option we are looking for is rendered by ng-select.
